nni/retiarii/graph.py

```diff
@@ -211,6 +211,28 @@ def get_node_by_name(self, node_name: str) -> 'Node':
             return matched_nodes[0]
         else:
             return None
+    
+    def get_node_by_python_name(self, python_name: str) -> 'Node':
+        """
+        Traverse all the nodes to find the matched node with the given python_name.
+        """
+        matched_nodes = []
+        for graph in self.graphs.values():
+            nodes = graph.get_nodes_by_python_name(python_name)
+            matched_nodes.extend(nodes)
+        # assert len(matched_nodes) <= 1
+        if matched_nodes:
+            return matched_nodes[0]
+        else:
+            return None
+
+    def get_layerchoice_nodes(self) -> List['Node']:
```

Comment:
Should not have `layerchoice` specific logic in graph. Suggest move it to graphgen.